src/core/amount.js

```diff
@@ -729,7 +705,9 @@ Amount.prototype.parse_native = function(j) {
 // Requires _currency to be set!
 Amount.prototype.parse_value = function(j) {
   this._is_native = false;
-  this._set_value(new BigNumber(j), BigNumber.ROUND_DOWN);
+  const newValue = new IOUValue(
+    new IOUValue(j).toPrecision(16, Value.getBNRoundDown()));
```

Comment:
Yeah it's a close call and I see your point. The BigNumber class does have the toPrecision function so it seems plausible that the Value could do it too. I'm going to go with the constructor option as I agree it's better to not expose the toPrecision method.